src/Host.cpp

```diff
@@ -369,6 +369,28 @@ void Host::resetProfile()
     qDebug()<<"resetProfile() DONE";
 }
 
+// Saves profile to disk - does not save items dirty in the editor, however.
+// returns true if successful or false+error message if not
+std::tuple<bool, QString, QString, QString, QString> Host::saveProfile()
```

Comment:
This is intended to be used by the C++ core to save profiles as well - I'll take your feedback onboard.